plugins/ua/universalarchive.plugin.zsh

```diff
@@ -0,0 +1,59 @@
+alias ua='universalarchive'
+alias ax='universalarchive tar.xz '
+
+# exclusively POSIX shell syntax is used
+
+universalarchive() { # compress a file or folder
+  local file
+  local parent
+  local name
+
+  local type="$1"
+  local input="$2"
+
+  if [ -f "$input" ]; then
+    if [ "$#" -eq 2 ]; then
+      name="${input%.[^.]*}"
+    else
+      file="$(realpath "$input")" 
+      parent="${file%/*}" 
+      name="${parent##*/}"
+    fi
+  elif [ -d "$input" ]; then
+    name="${input%/*}"
+  else
+    echo "universalarchive(): archive a file or directory."
+    echo "Usage:   ua <archive type> <files>"
+    echo "Error: Please give a valid file or directory name!"
+  fi
+
+  if [ -f "${name}.${type}" ]; then
+    name=$(mktemp --dry-run "./${name}_XXX.${type}")
```

Comment:
`--dry-run` is exclusively a GNU mktemp feature:

[1] http://man.openbsd.org/mktemp.1
[2] https://www.freebsd.org/cgi/man.cgi?query=mktemp
[3] https://docs.oracle.com/cd/E23823_01/html/816-5165/mktemp-1.html#scrolltoc

[See this post](url) for the rest of mktemp docs across platforms.